internal/cluster/cluster.go

```diff
@@ -211,6 +216,64 @@ func (m *Manager) DeregisterCluster(ctx context.Context, c *common.Cluster) erro
 	return err
 }
 
+func (m *Manager) reportValidationFailedMetrics(ctx context.Context, c *common.Cluster, ocpVersion, emailDomain string) error {
+	log := logutil.FromContext(ctx, m.log)
+	var validationRes validationsStatus
+	if c.ValidationsInfo != "" {
+		if err := json.Unmarshal([]byte(c.ValidationsInfo), &validationRes); err != nil {
+			log.WithError(err).Errorf("Failed to unmarshal validations info from cluster %s", *c.ID)
+			return err
+		}
+	}
+	for _, vRes := range validationRes {
+		for _, v := range vRes {
+			if v.Status == ValidationFailure {
+				m.metricAPI.ClusterValidationFailed(ocpVersion, emailDomain, models.ClusterValidationID(v.ID))
+			}
+		}
+	}
+	return nil
+}
+
+func (m *Manager) reportValidationStatusChanged(ctx context.Context, c *common.Cluster,
+	newValidationRes, currentValidationRes validationsStatus) {
+	for vCategory, vRes := range currentValidationRes {
+		for i, v := range vRes {
+			if newValidationRes[vCategory][i].Status == ValidationFailure && v.Status == ValidationSuccess {
+				m.metricAPI.ClusterValidationChanged(c.OpenshiftVersion, c.EmailDomain, models.ClusterValidationID(v.ID))
+				eventMsg := fmt.Sprintf("Cluster validation '%s' that used to succeed is now failing", v.ID)
+				m.eventsHandler.AddEvent(ctx, *c.ID, nil, models.EventSeverityWarning, eventMsg, time.Now())
+			}
+			if newValidationRes[vCategory][i].Status == ValidationSuccess && v.Status == ValidationFailure {
+				eventMsg := fmt.Sprintf("Cluster validation '%s' is now fixed", v.ID)
+				m.eventsHandler.AddEvent(ctx, *c.ID, nil, models.EventSeverityInfo, eventMsg, time.Now())
+			}
+		}
+	}
+}
+
+func (m *Manager) checkValidationChanged(ctx context.Context, c *common.Cluster, newValidationRes validationsStatus) (validationsStatus, bool, error) {
+	var currentValidationRes validationsStatus
+	if c.ValidationsInfo != "" {
+		if err := json.Unmarshal([]byte(c.ValidationsInfo), &currentValidationRes); err != nil {
+			return validationsStatus{}, false, errors.Wrapf(err, "Failed to unmarshal validations info from cluster %s", *c.ID)
+		}
+	}
```

Comment:
Consider moving this to a seperate function, seems that `checkValidationChanged` is returning 3 things because it's also doing `getNewValidationResults`

Reply:
> it's also doing getNewValidationResults

It is not, it is getting it as a parameter.
We had a meeting about it with @ori-amizur and this is how we decided to implement https://github.com/openshift/assisted-service/pull/1234

Reply:
OK, so why is this method returning validationsStatus?

Reply:
I see the confusion, I meant `getCurrentValidationResults`